### PR TITLE
editTopic UI component

### PR DESCRIPTION
**Pull request by ashatna**
add url to item urls
implement put and delete methods in item views
create UI components: AddItemDialog, EditItemDialog, ItemServices, EditTopicDialog
modify UI components: ItemLink, TopicItemList, TopicDialog, TopicServices, TopicView
modify index.js file
implement put method in topic views

## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/browse/LEAR-60)

## Code reviewers

- [ ] @ashatna
- [ ] @ChristinaPastushchak
- [ ] @dotiurin
- [ ] @iryn34
- [x] @tarassito
- [ ] @VasilHamulyak
- [x] @VitalyiKhlivnyi
- [ ] @YurijH

### Second Level Review

- [ ] @ushevtsiv
- [x] @lhalam
- [ ] @ssOleg

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by VitalyiKhlivnyi — DISMISSED** [review]

**Review by VitalyiKhlivnyi — DISMISSED** [review]

File: eventually/item/views.py [hunk 1]
```diff
@@ -96,3 +98,79 @@ def post(self, request, curriculum_id, topic_id):
...
+        :return: response with status code 200 when topic was successfully deleted or response with
+                 403 or 404 failed status code.
+        :rtype: `HttpResponse object."""
+
+        user = request.user
+        item = Item.get_by_id(item_id)
+        topic = Topic.get_by_id(topic_id)
+        if not item:
+            return RESPONSE_404_OBJECT_NOT_FOUND
+        if user in topic.mentors.all():
```
Comment (lhalam) [inline eventually/item/views.py]: if topic None?

File: eventually/static/src/containers/topicItemList/EditItemDialog.js [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+                        onChange={this.handleDescription}
+                        errorText={this.state.messageDescription}
+                        multiLine={true}
+                        rowsMax={4}
+                        fullWidth={true} />
+                    <SelectField
+                        floatingLabelText="Select form"
+                        value={this.state.form}
+                        onChange={this.handleChange}
+                    >
```
Comment (lhalam) [inline eventually/static/src/containers/topicItemList/EditItemDialog.js]: ` onChange={this.handleChange}
                    >`
to
` onChange={this.handleChange}>`

File: eventually/static/src/containers/topicItemList/AddItemDialog.js [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    };
+
+    handleName = event => {
+        const regex = /^.{4,255}$/;
+        if(regex.test(event.target.value) === true ) {
+            this.setState({
+                messageName: '',
+                name: event.target.value,
+                nameIsValid: true,
+            });
```
Comment (tarassito) [inline eventually/static/src/containers/topicItemList/AddItemDialog.js:61]: в стейті немає messageName , nameIsValid
воно так працює?

File: eventually/item/views.py (hunk 1 above)
Comment (ashatna) [reply eventually/item/views.py]: done!

File: eventually/static/src/containers/topicItemList/AddItemDialog.js (hunk 3 above)
Comment (ashatna) [reply eventually/static/src/containers/topicItemList/AddItemDialog.js:61]: працювало )
але все ж додала то до стейту.

File: eventually/static/src/containers/topicItemList/EditItemDialog.js (hunk 2 above)
Comment (ashatna) [reply eventually/static/src/containers/topicItemList/EditItemDialog.js]: done

**Review by dotiurin — DISMISSED** [review]

**Review by lhalam — DISMISSED** [review]

**Review by VitalyiKhlivnyi — APPROVED** [review]

**Pull request merged**